example.html

```diff
@@ -1,43 +1,73 @@
-      closeOnUnload: true // See https://support.ably.io/solution/articles/3000059875-why-don-t-presence-members-leave-as-soon-as-i-close-a-tab-
+  <script>
+    const apiKey = "YOUR_ABLY_API_KEY"; // Replace with your API key
+    const randomId =
+      Math.random().toString(36).substring(2, 10) +
+      Math.random().toString(36).substring(2, 10);
+    const realtime = new Ably.Realtime.Promise({
+      key: apiKey,
+      clientId: randomId, // Your ID in the presence set
     });
 
-    /* Enter the presence set of the 'chatroom' channel */
-    var channel = realtime.channels.get('chatroom');
-    channel.attach(function(err) {
-      if(err) { return console.error("Error attaching to the channel"); }
-      console.log('We are now attached to the channel');
-      channel.presence.enter('hello', function(err) {
-        if(err) { return console.error("Error entering presence"); }
-        console.log('We are now successfully present');
+    async function doPresence() {
+      // Connect to Ably
+      await realtime.connection.once("connected");
+      console.log("Connected to Ably!");
+
+      // Attach to the "chatroom" channel
+      const channel = realtime.channels.get("chatroom");
+      await channel.attach((err) => {
+        if (err) {
+          return console.error("Error attaching to the channel.");
+        }
+      });
+
+      // Enter the presence set of the "chatroom" channel
+      await channel.presence.enter("hello", (err) => {
+        if (err) {
+          return console.error("Error entering presence set.");
+        }
+        console.log("This client has entered the presence set.");
       });
 
-      /* Every time the presence set changes, show the new set */
-      channel.presence.subscribe(function(presenceMsg) {
-        console.log('Received a ' + presenceMsg.action + ' from ' + presenceMsg.clientId);
-        channel.presence.get(function(err, members) {
-          if(err) { return console.log("Error fetching presence data: " + err); }
-          document.getElementById('presence-set').innerHTML = members.map(function(member) {
-            return '<li>' + member.clientId + '</li>';
-          }).join();
+      // Subscribe to the presence set to receive updates
+      await channel.presence.subscribe((presenceMessage) => {
+        console.log(
+          "Presence update: " +
+            presenceMessage.action +
+            " from " +
+            presenceMessage.clientId
+        );
+        // Update the list of channel members when the presence set changes
+        channel.presence.get((err, members) => {
+          if (err) {
+            return console.error(`Error retrieving presence data: ${err}`);
+          }
+          document.getElementById("presence-set").innerHTML = members
+            .map((member) => {
+              return "<li>" + member.clientId + "</li>";
+            })
+            .join("");
```

Comment:
Please use String literals so the method reads easier.

```suggestion
          document.getElementById("#presence-set").innerHTML = members
            .map(({clientid}) => `<li>${clientId}</li>`)
            .join("");
```

Reply:
👍